pkg/kvstore/etcd.go

```diff
@@ -1343,6 +1343,23 @@ func (e *etcdClient) ListAndWatch(ctx context.Context, name, prefix string, chan
 	return w
 }
 
+// GetSvcNamespace returns the service name and namespace for the given address.
+// If the given address is not parseable or it is not the format
+// '<protocol://><name>.<namespace>[optional]', returns an error.
+func GetSvcNamespace(address string) (string, string, error) {
```

Comment:
Call this `SplitK8sServiceURL` or similar to indicate that this is a string splitting function.